{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Create SRA submission sheet\n",
    "This Python Jupyter notebook creates a BioSample submission sheet for the Sequence Read Archive:\n",
    "\n",
    "First, import Python modules:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import datetime\n",
    "import glob\n",
    "import itertools\n",
    "import os\n",
    "\n",
    "import natsort\n",
    "\n",
    "import pandas as pd\n",
    "\n",
    "import yaml"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Display full columns of data frames:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "pd.set_option('display.max_colwidth', None)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Read the configuration for the uploads:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open('upload_config.yaml') as f:\n",
    "    config = yaml.safe_load(f)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Read the Illumina runs to submit:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Reading Illumina runs from barcode_runs.csv\n",
      "Here are the first few entries in the Illumina runs:\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>date</th>\n",
       "      <th>experiment</th>\n",
       "      <th>library</th>\n",
       "      <th>antibody</th>\n",
       "      <th>concentration</th>\n",
       "      <th>sort_bin</th>\n",
       "      <th>selection</th>\n",
       "      <th>sample</th>\n",
       "      <th>experiment_type</th>\n",
       "      <th>number_cells</th>\n",
       "      <th>frac_escape</th>\n",
       "      <th>R1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>220118</td>\n",
       "      <td>beta_19-26</td>\n",
       "      <td>lib1</td>\n",
       "      <td>none</td>\n",
       "      <td>0</td>\n",
       "      <td>ref</td>\n",
       "      <td>reference</td>\n",
       "      <td>beta_19-26-none-0-ref</td>\n",
       "      <td>ab_selection</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/shared/ngs/illumina/agreaney/220124_D00300_1402_BHNKFCBCX3/Unaligned/Project_agreaney/beta_19-26_lib1_ref*R1*.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>220118</td>\n",
       "      <td>beta_19-26</td>\n",
       "      <td>lib2</td>\n",
       "      <td>none</td>\n",
       "      <td>0</td>\n",
       "      <td>ref</td>\n",
       "      <td>reference</td>\n",
       "      <td>beta_19-26-none-0-ref</td>\n",
       "      <td>ab_selection</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/shared/ngs/illumina/agreaney/220124_D00300_1402_BHNKFCBCX3/Unaligned/Project_agreaney/beta_19-26_lib2_ref*R1*.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>220118</td>\n",
       "      <td>beta_19</td>\n",
       "      <td>lib1</td>\n",
       "      <td>mosaic_6848</td>\n",
       "      <td>3125</td>\n",
       "      <td>abneg</td>\n",
       "      <td>escape</td>\n",
       "      <td>beta_19-mosaic_6848-3125-abneg</td>\n",
       "      <td>ab_selection</td>\n",
       "      <td>875492.0</td>\n",
       "      <td>0.096</td>\n",
       "      <td>/shared/ngs/illumina/agreaney/220124_D00300_1402_BHNKFCBCX3/Unaligned/Project_agreaney/beta_19_lib1_abneg*R1*.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>220118</td>\n",
       "      <td>beta_19</td>\n",
       "      <td>lib2</td>\n",
       "      <td>mosaic_6848</td>\n",
       "      <td>3125</td>\n",
       "      <td>abneg</td>\n",
       "      <td>escape</td>\n",
       "      <td>beta_19-mosaic_6848-3125-abneg</td>\n",
       "      <td>ab_selection</td>\n",
       "      <td>1002826.0</td>\n",
       "      <td>0.136</td>\n",
       "      <td>/shared/ngs/illumina/agreaney/220124_D00300_1402_BHNKFCBCX3/Unaligned/Project_agreaney/beta_19_lib2_abneg*R1*.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>220118</td>\n",
       "      <td>beta_20</td>\n",
       "      <td>lib1</td>\n",
       "      <td>mosaic_6849</td>\n",
       "      <td>3125</td>\n",
       "      <td>abneg</td>\n",
       "      <td>escape</td>\n",
       "      <td>beta_20-mosaic_6849-3125-abneg</td>\n",
       "      <td>ab_selection</td>\n",
       "      <td>957229.0</td>\n",
       "      <td>0.105</td>\n",
       "      <td>/shared/ngs/illumina/agreaney/220124_D00300_1402_BHNKFCBCX3/Unaligned/Project_agreaney/beta_20_lib1_abneg*R1*.fastq.gz</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     date  experiment library     antibody  concentration sort_bin  selection  \\\n",
       "0  220118  beta_19-26    lib1         none              0      ref  reference   \n",
       "1  220118  beta_19-26    lib2         none              0      ref  reference   \n",
       "2  220118     beta_19    lib1  mosaic_6848           3125    abneg     escape   \n",
       "3  220118     beta_19    lib2  mosaic_6848           3125    abneg     escape   \n",
       "4  220118     beta_20    lib1  mosaic_6849           3125    abneg     escape   \n",
       "\n",
       "                           sample experiment_type  number_cells  frac_escape  \\\n",
       "0           beta_19-26-none-0-ref    ab_selection           NaN          NaN   \n",
       "1           beta_19-26-none-0-ref    ab_selection           NaN          NaN   \n",
       "2  beta_19-mosaic_6848-3125-abneg    ab_selection      875492.0        0.096   \n",
       "3  beta_19-mosaic_6848-3125-abneg    ab_selection     1002826.0        0.136   \n",
       "4  beta_20-mosaic_6849-3125-abneg    ab_selection      957229.0        0.105   \n",
       "\n",
       "                                                                                                                        R1  \n",
       "0  /shared/ngs/illumina/agreaney/220124_D00300_1402_BHNKFCBCX3/Unaligned/Project_agreaney/beta_19-26_lib1_ref*R1*.fastq.gz  \n",
       "1  /shared/ngs/illumina/agreaney/220124_D00300_1402_BHNKFCBCX3/Unaligned/Project_agreaney/beta_19-26_lib2_ref*R1*.fastq.gz  \n",
       "2   /shared/ngs/illumina/agreaney/220124_D00300_1402_BHNKFCBCX3/Unaligned/Project_agreaney/beta_19_lib1_abneg*R1*.fastq.gz  \n",
       "3   /shared/ngs/illumina/agreaney/220124_D00300_1402_BHNKFCBCX3/Unaligned/Project_agreaney/beta_19_lib2_abneg*R1*.fastq.gz  \n",
       "4   /shared/ngs/illumina/agreaney/220124_D00300_1402_BHNKFCBCX3/Unaligned/Project_agreaney/beta_20_lib1_abneg*R1*.fastq.gz  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(f\"Reading Illumina runs from {config['barcode_runs']}\")\n",
    "\n",
    "illumina_runs = pd.read_csv(config['barcode_runs'])\n",
    "\n",
    "print('Here are the first few entries in the Illumina runs:')\n",
    "illumina_runs.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Next make submission entries for the barcode runs, initially in \"tidy\" format with one FASTQ file per row:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Here are the first few submission entries in tidy format:\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>biosample_accession</th>\n",
       "      <th>library_ID</th>\n",
       "      <th>title</th>\n",
       "      <th>library_strategy</th>\n",
       "      <th>library_source</th>\n",
       "      <th>library_selection</th>\n",
       "      <th>library_layout</th>\n",
       "      <th>platform</th>\n",
       "      <th>instrument_model</th>\n",
       "      <th>design_description</th>\n",
       "      <th>filetype</th>\n",
       "      <th>filename_fullpath</th>\n",
       "      <th>filename</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SAMN26315988</td>\n",
       "      <td>lib1_beta_19-26-none-0-ref</td>\n",
       "      <td>SARS-CoV-2 RBD mutational antig Illumina barcode sequencing for beta_19-26-none-0-ref</td>\n",
       "      <td>AMPLICON</td>\n",
       "      <td>SYNTHETIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>single</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina HiSeq 2500</td>\n",
       "      <td>PCR of barcodes from RBD variants</td>\n",
       "      <td>fastq</td>\n",
       "      <td>/shared/ngs/illumina/agreaney/220124_D00300_1402_BHNKFCBCX3/Unaligned/Project_agreaney/beta_19-26_lib1_ref-3_S3_R1_001.fastq.gz</td>\n",
       "      <td>beta_19-26_lib1_ref-3_S3_R1_001.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SAMN26315988</td>\n",
       "      <td>lib1_beta_19-26-none-0-ref</td>\n",
       "      <td>SARS-CoV-2 RBD mutational antig Illumina barcode sequencing for beta_19-26-none-0-ref</td>\n",
       "      <td>AMPLICON</td>\n",
       "      <td>SYNTHETIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>single</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina HiSeq 2500</td>\n",
       "      <td>PCR of barcodes from RBD variants</td>\n",
       "      <td>fastq</td>\n",
       "      <td>/shared/ngs/illumina/agreaney/220124_D00300_1402_BHNKFCBCX3/Unaligned/Project_agreaney/beta_19-26_lib1_ref-2_S2_R1_001.fastq.gz</td>\n",
       "      <td>beta_19-26_lib1_ref-2_S2_R1_001.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SAMN26315988</td>\n",
       "      <td>lib1_beta_19-26-none-0-ref</td>\n",
       "      <td>SARS-CoV-2 RBD mutational antig Illumina barcode sequencing for beta_19-26-none-0-ref</td>\n",
       "      <td>AMPLICON</td>\n",
       "      <td>SYNTHETIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>single</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina HiSeq 2500</td>\n",
       "      <td>PCR of barcodes from RBD variants</td>\n",
       "      <td>fastq</td>\n",
       "      <td>/shared/ngs/illumina/agreaney/220124_D00300_1402_BHNKFCBCX3/Unaligned/Project_agreaney/beta_19-26_lib1_ref-4_S4_R1_001.fastq.gz</td>\n",
       "      <td>beta_19-26_lib1_ref-4_S4_R1_001.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SAMN26315988</td>\n",
       "      <td>lib1_beta_19-26-none-0-ref</td>\n",
       "      <td>SARS-CoV-2 RBD mutational antig Illumina barcode sequencing for beta_19-26-none-0-ref</td>\n",
       "      <td>AMPLICON</td>\n",
       "      <td>SYNTHETIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>single</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina HiSeq 2500</td>\n",
       "      <td>PCR of barcodes from RBD variants</td>\n",
       "      <td>fastq</td>\n",
       "      <td>/shared/ngs/illumina/agreaney/220124_D00300_1402_BHNKFCBCX3/Unaligned/Project_agreaney/beta_19-26_lib1_ref-1_S1_R1_001.fastq.gz</td>\n",
       "      <td>beta_19-26_lib1_ref-1_S1_R1_001.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SAMN26315988</td>\n",
       "      <td>lib2_beta_19-26-none-0-ref</td>\n",
       "      <td>SARS-CoV-2 RBD mutational antig Illumina barcode sequencing for beta_19-26-none-0-ref</td>\n",
       "      <td>AMPLICON</td>\n",
       "      <td>SYNTHETIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>single</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina HiSeq 2500</td>\n",
       "      <td>PCR of barcodes from RBD variants</td>\n",
       "      <td>fastq</td>\n",
       "      <td>/shared/ngs/illumina/agreaney/220124_D00300_1402_BHNKFCBCX3/Unaligned/Project_agreaney/beta_19-26_lib2_ref-2_S6_R1_001.fastq.gz</td>\n",
       "      <td>beta_19-26_lib2_ref-2_S6_R1_001.fastq.gz</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  biosample_accession                  library_ID  \\\n",
       "0        SAMN26315988  lib1_beta_19-26-none-0-ref   \n",
       "1        SAMN26315988  lib1_beta_19-26-none-0-ref   \n",
       "2        SAMN26315988  lib1_beta_19-26-none-0-ref   \n",
       "3        SAMN26315988  lib1_beta_19-26-none-0-ref   \n",
       "4        SAMN26315988  lib2_beta_19-26-none-0-ref   \n",
       "\n",
       "                                                                                   title  \\\n",
       "0  SARS-CoV-2 RBD mutational antig Illumina barcode sequencing for beta_19-26-none-0-ref   \n",
       "1  SARS-CoV-2 RBD mutational antig Illumina barcode sequencing for beta_19-26-none-0-ref   \n",
       "2  SARS-CoV-2 RBD mutational antig Illumina barcode sequencing for beta_19-26-none-0-ref   \n",
       "3  SARS-CoV-2 RBD mutational antig Illumina barcode sequencing for beta_19-26-none-0-ref   \n",
       "4  SARS-CoV-2 RBD mutational antig Illumina barcode sequencing for beta_19-26-none-0-ref   \n",
       "\n",
       "  library_strategy library_source library_selection library_layout  platform  \\\n",
       "0         AMPLICON      SYNTHETIC               PCR         single  ILLUMINA   \n",
       "1         AMPLICON      SYNTHETIC               PCR         single  ILLUMINA   \n",
       "2         AMPLICON      SYNTHETIC               PCR         single  ILLUMINA   \n",
       "3         AMPLICON      SYNTHETIC               PCR         single  ILLUMINA   \n",
       "4         AMPLICON      SYNTHETIC               PCR         single  ILLUMINA   \n",
       "\n",
       "      instrument_model                 design_description filetype  \\\n",
       "0  Illumina HiSeq 2500  PCR of barcodes from RBD variants    fastq   \n",
       "1  Illumina HiSeq 2500  PCR of barcodes from RBD variants    fastq   \n",
       "2  Illumina HiSeq 2500  PCR of barcodes from RBD variants    fastq   \n",
       "3  Illumina HiSeq 2500  PCR of barcodes from RBD variants    fastq   \n",
       "4  Illumina HiSeq 2500  PCR of barcodes from RBD variants    fastq   \n",
       "\n",
       "                                                                                                                 filename_fullpath  \\\n",
       "0  /shared/ngs/illumina/agreaney/220124_D00300_1402_BHNKFCBCX3/Unaligned/Project_agreaney/beta_19-26_lib1_ref-3_S3_R1_001.fastq.gz   \n",
       "1  /shared/ngs/illumina/agreaney/220124_D00300_1402_BHNKFCBCX3/Unaligned/Project_agreaney/beta_19-26_lib1_ref-2_S2_R1_001.fastq.gz   \n",
       "2  /shared/ngs/illumina/agreaney/220124_D00300_1402_BHNKFCBCX3/Unaligned/Project_agreaney/beta_19-26_lib1_ref-4_S4_R1_001.fastq.gz   \n",
       "3  /shared/ngs/illumina/agreaney/220124_D00300_1402_BHNKFCBCX3/Unaligned/Project_agreaney/beta_19-26_lib1_ref-1_S1_R1_001.fastq.gz   \n",
       "4  /shared/ngs/illumina/agreaney/220124_D00300_1402_BHNKFCBCX3/Unaligned/Project_agreaney/beta_19-26_lib2_ref-2_S6_R1_001.fastq.gz   \n",
       "\n",
       "                                   filename  \n",
       "0  beta_19-26_lib1_ref-3_S3_R1_001.fastq.gz  \n",
       "1  beta_19-26_lib1_ref-2_S2_R1_001.fastq.gz  \n",
       "2  beta_19-26_lib1_ref-4_S4_R1_001.fastq.gz  \n",
       "3  beta_19-26_lib1_ref-1_S1_R1_001.fastq.gz  \n",
       "4  beta_19-26_lib2_ref-2_S6_R1_001.fastq.gz  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "submissions_tidy = (\n",
    "    illumina_runs\n",
    "    .assign(\n",
    "        biosample_accession=lambda x: config['biosample_accession'],\n",
    "        library_ID=lambda x: x['library'] + '_' + x['sample'],\n",
    "        title=lambda x: 'SARS-CoV-2 RBD mutational antig Illumina barcode sequencing for ' + x['sample'],\n",
    "        library_strategy='AMPLICON',\n",
    "        library_source='SYNTHETIC',\n",
    "        library_selection='PCR',\n",
    "        library_layout='single',\n",
    "        platform='ILLUMINA',\n",
    "        instrument_model='Illumina HiSeq 2500',\n",
    "        design_description='PCR of barcodes from RBD variants',\n",
    "        filetype='fastq',\n",
    "        filename_fullpath=lambda x: x['R1'].str.split(';')\n",
    "                                    .map(lambda flist: list(itertools.chain.from_iterable(glob.glob(f) for f in flist))),       \n",
    "        )\n",
    "    .explode('filename_fullpath')\n",
    "    .assign(filename_fullpath=lambda x: x['filename_fullpath'].str.strip(),\n",
    "            filename=lambda x: x['filename_fullpath'].map(os.path.basename))\n",
    "    .drop(columns=illumina_runs.columns)\n",
    "    .reset_index(drop=True)\n",
    "    )\n",
    "\n",
    "assert submissions_tidy['filename_fullpath'].map(os.path.isfile).all()\n",
    "\n",
    "print('Here are the first few submission entries in tidy format:')\n",
    "submissions_tidy.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "For the actual submission, we need a \"wide\" data frame that for each unique `sample_name` / `library_ID` gives all of the files each in different columns.\n",
    "These should be files without the full path.\n",
    "\n",
    "First, look at how many files there are for each sample / library:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>biosample_accession</th>\n",
       "      <th>library_ID</th>\n",
       "      <th>n_files</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SAMN26315988</td>\n",
       "      <td>lib1_beta_19-26-none-0-ref</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SAMN26315988</td>\n",
       "      <td>lib1_beta_27-34-none-0-ref</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SAMN26315988</td>\n",
       "      <td>lib2_beta_27-34-none-0-ref</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SAMN26315988</td>\n",
       "      <td>lib2_beta_19-26-none-0-ref</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SAMN26315988</td>\n",
       "      <td>lib1_beta_22-mosaic_6851-3125-abneg</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>SAMN26315988</td>\n",
       "      <td>lib1_beta_23-mosaic_6852-3125-abneg</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>SAMN26315988</td>\n",
       "      <td>lib2_beta_33-NHP_mosaic_0Z5-1000-abneg</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>SAMN26315988</td>\n",
       "      <td>lib2_beta_32-NHP_mosaic_AO2-1000-abneg</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>SAMN26315988</td>\n",
       "      <td>lib2_beta_31-NHP_mosaic_MA292-1000-abneg</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>SAMN26315988</td>\n",
       "      <td>lib2_beta_30-homotypic_6885-5000-abneg</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>SAMN26315988</td>\n",
       "      <td>lib2_beta_29-homotypic_6884-5000-abneg</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>SAMN26315988</td>\n",
       "      <td>lib2_beta_28-homotypic_6882-5000-abneg</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>SAMN26315988</td>\n",
       "      <td>lib2_beta_27-homotypic_6880-5000-abneg</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>SAMN26315988</td>\n",
       "      <td>lib1_beta_20-mosaic_6849-3125-abneg</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>SAMN26315988</td>\n",
       "      <td>lib2_beta_26-homotypic_6883-7812-abneg</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>SAMN26315988</td>\n",
       "      <td>lib2_beta_25-homotypic_6881-7812-abneg</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>SAMN26315988</td>\n",
       "      <td>lib2_beta_24-mosaic_6853-3125-abneg</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>SAMN26315988</td>\n",
       "      <td>lib2_beta_23-mosaic_6852-3125-abneg</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>SAMN26315988</td>\n",
       "      <td>lib2_beta_22-mosaic_6851-3125-abneg</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>SAMN26315988</td>\n",
       "      <td>lib2_beta_21-mosaic_6850-3125-abneg</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>SAMN26315988</td>\n",
       "      <td>lib2_beta_20-mosaic_6849-3125-abneg</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>SAMN26315988</td>\n",
       "      <td>lib2_beta_19-mosaic_6848-3125-abneg</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>SAMN26315988</td>\n",
       "      <td>lib1_beta_19-mosaic_6848-3125-abneg</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>SAMN26315988</td>\n",
       "      <td>lib1_beta_34-NHP_mosaic_AT5-1000-abneg</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>SAMN26315988</td>\n",
       "      <td>lib1_beta_33-NHP_mosaic_0Z5-1000-abneg</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>SAMN26315988</td>\n",
       "      <td>lib1_beta_32-NHP_mosaic_AO2-1000-abneg</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>SAMN26315988</td>\n",
       "      <td>lib1_beta_31-NHP_mosaic_MA292-1000-abneg</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>SAMN26315988</td>\n",
       "      <td>lib1_beta_30-homotypic_6885-5000-abneg</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>SAMN26315988</td>\n",
       "      <td>lib1_beta_29-homotypic_6884-5000-abneg</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>SAMN26315988</td>\n",
       "      <td>lib1_beta_28-homotypic_6882-5000-abneg</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>SAMN26315988</td>\n",
       "      <td>lib1_beta_27-homotypic_6880-5000-abneg</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>SAMN26315988</td>\n",
       "      <td>lib1_beta_21-mosaic_6850-3125-abneg</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>SAMN26315988</td>\n",
       "      <td>lib1_beta_26-homotypic_6883-7812-abneg</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>SAMN26315988</td>\n",
       "      <td>lib1_beta_25-homotypic_6881-7812-abneg</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>SAMN26315988</td>\n",
       "      <td>lib1_beta_24-mosaic_6853-3125-abneg</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>SAMN26315988</td>\n",
       "      <td>lib2_beta_34-NHP_mosaic_AT5-1000-abneg</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   biosample_accession                                library_ID  n_files\n",
       "0         SAMN26315988                lib1_beta_19-26-none-0-ref        4\n",
       "1         SAMN26315988                lib1_beta_27-34-none-0-ref        4\n",
       "2         SAMN26315988                lib2_beta_27-34-none-0-ref        4\n",
       "3         SAMN26315988                lib2_beta_19-26-none-0-ref        4\n",
       "4         SAMN26315988       lib1_beta_22-mosaic_6851-3125-abneg        1\n",
       "5         SAMN26315988       lib1_beta_23-mosaic_6852-3125-abneg        1\n",
       "6         SAMN26315988    lib2_beta_33-NHP_mosaic_0Z5-1000-abneg        1\n",
       "7         SAMN26315988    lib2_beta_32-NHP_mosaic_AO2-1000-abneg        1\n",
       "8         SAMN26315988  lib2_beta_31-NHP_mosaic_MA292-1000-abneg        1\n",
       "9         SAMN26315988    lib2_beta_30-homotypic_6885-5000-abneg        1\n",
       "10        SAMN26315988    lib2_beta_29-homotypic_6884-5000-abneg        1\n",
       "11        SAMN26315988    lib2_beta_28-homotypic_6882-5000-abneg        1\n",
       "12        SAMN26315988    lib2_beta_27-homotypic_6880-5000-abneg        1\n",
       "13        SAMN26315988       lib1_beta_20-mosaic_6849-3125-abneg        1\n",
       "14        SAMN26315988    lib2_beta_26-homotypic_6883-7812-abneg        1\n",
       "15        SAMN26315988    lib2_beta_25-homotypic_6881-7812-abneg        1\n",
       "16        SAMN26315988       lib2_beta_24-mosaic_6853-3125-abneg        1\n",
       "17        SAMN26315988       lib2_beta_23-mosaic_6852-3125-abneg        1\n",
       "18        SAMN26315988       lib2_beta_22-mosaic_6851-3125-abneg        1\n",
       "19        SAMN26315988       lib2_beta_21-mosaic_6850-3125-abneg        1\n",
       "20        SAMN26315988       lib2_beta_20-mosaic_6849-3125-abneg        1\n",
       "21        SAMN26315988       lib2_beta_19-mosaic_6848-3125-abneg        1\n",
       "22        SAMN26315988       lib1_beta_19-mosaic_6848-3125-abneg        1\n",
       "23        SAMN26315988    lib1_beta_34-NHP_mosaic_AT5-1000-abneg        1\n",
       "24        SAMN26315988    lib1_beta_33-NHP_mosaic_0Z5-1000-abneg        1\n",
       "25        SAMN26315988    lib1_beta_32-NHP_mosaic_AO2-1000-abneg        1\n",
       "26        SAMN26315988  lib1_beta_31-NHP_mosaic_MA292-1000-abneg        1\n",
       "27        SAMN26315988    lib1_beta_30-homotypic_6885-5000-abneg        1\n",
       "28        SAMN26315988    lib1_beta_29-homotypic_6884-5000-abneg        1\n",
       "29        SAMN26315988    lib1_beta_28-homotypic_6882-5000-abneg        1\n",
       "30        SAMN26315988    lib1_beta_27-homotypic_6880-5000-abneg        1\n",
       "31        SAMN26315988       lib1_beta_21-mosaic_6850-3125-abneg        1\n",
       "32        SAMN26315988    lib1_beta_26-homotypic_6883-7812-abneg        1\n",
       "33        SAMN26315988    lib1_beta_25-homotypic_6881-7812-abneg        1\n",
       "34        SAMN26315988       lib1_beta_24-mosaic_6853-3125-abneg        1\n",
       "35        SAMN26315988    lib2_beta_34-NHP_mosaic_AT5-1000-abneg        1"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "(submissions_tidy\n",
    " .groupby(['biosample_accession', 'library_ID'])\n",
    " .aggregate(n_files=pd.NamedAgg('filename_fullpath', 'count'))\n",
    " .sort_values('n_files', ascending=False)\n",
    " .reset_index()\n",
    " )"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now make the wide submission data frame.\n",
    "Note we keep only the filename column with the path lacking the full directory information:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Here are the first few submission entries in wide format:\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>biosample_accession</th>\n",
       "      <th>library_ID</th>\n",
       "      <th>title</th>\n",
       "      <th>library_strategy</th>\n",
       "      <th>library_source</th>\n",
       "      <th>library_selection</th>\n",
       "      <th>library_layout</th>\n",
       "      <th>platform</th>\n",
       "      <th>instrument_model</th>\n",
       "      <th>design_description</th>\n",
       "      <th>filetype</th>\n",
       "      <th>filename</th>\n",
       "      <th>filename2</th>\n",
       "      <th>filename3</th>\n",
       "      <th>filename4</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SAMN26315988</td>\n",
       "      <td>lib1_beta_19-26-none-0-ref</td>\n",
       "      <td>SARS-CoV-2 RBD mutational antig Illumina barcode sequencing for beta_19-26-none-0-ref</td>\n",
       "      <td>AMPLICON</td>\n",
       "      <td>SYNTHETIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>single</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina HiSeq 2500</td>\n",
       "      <td>PCR of barcodes from RBD variants</td>\n",
       "      <td>fastq</td>\n",
       "      <td>beta_19-26_lib1_ref-3_S3_R1_001.fastq.gz</td>\n",
       "      <td>beta_19-26_lib1_ref-2_S2_R1_001.fastq.gz</td>\n",
       "      <td>beta_19-26_lib1_ref-4_S4_R1_001.fastq.gz</td>\n",
       "      <td>beta_19-26_lib1_ref-1_S1_R1_001.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SAMN26315988</td>\n",
       "      <td>lib2_beta_19-26-none-0-ref</td>\n",
       "      <td>SARS-CoV-2 RBD mutational antig Illumina barcode sequencing for beta_19-26-none-0-ref</td>\n",
       "      <td>AMPLICON</td>\n",
       "      <td>SYNTHETIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>single</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina HiSeq 2500</td>\n",
       "      <td>PCR of barcodes from RBD variants</td>\n",
       "      <td>fastq</td>\n",
       "      <td>beta_19-26_lib2_ref-2_S6_R1_001.fastq.gz</td>\n",
       "      <td>beta_19-26_lib2_ref-4_S8_R1_001.fastq.gz</td>\n",
       "      <td>beta_19-26_lib2_ref-1_S5_R1_001.fastq.gz</td>\n",
       "      <td>beta_19-26_lib2_ref-3_S7_R1_001.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SAMN26315988</td>\n",
       "      <td>lib1_beta_19-mosaic_6848-3125-abneg</td>\n",
       "      <td>SARS-CoV-2 RBD mutational antig Illumina barcode sequencing for beta_19-mosaic_6848-3125-abneg</td>\n",
       "      <td>AMPLICON</td>\n",
       "      <td>SYNTHETIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>single</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina HiSeq 2500</td>\n",
       "      <td>PCR of barcodes from RBD variants</td>\n",
       "      <td>fastq</td>\n",
       "      <td>beta_19_lib1_abneg_S9_R1_001.fastq.gz</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SAMN26315988</td>\n",
       "      <td>lib2_beta_19-mosaic_6848-3125-abneg</td>\n",
       "      <td>SARS-CoV-2 RBD mutational antig Illumina barcode sequencing for beta_19-mosaic_6848-3125-abneg</td>\n",
       "      <td>AMPLICON</td>\n",
       "      <td>SYNTHETIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>single</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina HiSeq 2500</td>\n",
       "      <td>PCR of barcodes from RBD variants</td>\n",
       "      <td>fastq</td>\n",
       "      <td>beta_19_lib2_abneg_S17_R1_001.fastq.gz</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SAMN26315988</td>\n",
       "      <td>lib1_beta_20-mosaic_6849-3125-abneg</td>\n",
       "      <td>SARS-CoV-2 RBD mutational antig Illumina barcode sequencing for beta_20-mosaic_6849-3125-abneg</td>\n",
       "      <td>AMPLICON</td>\n",
       "      <td>SYNTHETIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>single</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina HiSeq 2500</td>\n",
       "      <td>PCR of barcodes from RBD variants</td>\n",
       "      <td>fastq</td>\n",
       "      <td>beta_20_lib1_abneg_S10_R1_001.fastq.gz</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  biosample_accession                           library_ID  \\\n",
       "0        SAMN26315988           lib1_beta_19-26-none-0-ref   \n",
       "1        SAMN26315988           lib2_beta_19-26-none-0-ref   \n",
       "2        SAMN26315988  lib1_beta_19-mosaic_6848-3125-abneg   \n",
       "3        SAMN26315988  lib2_beta_19-mosaic_6848-3125-abneg   \n",
       "4        SAMN26315988  lib1_beta_20-mosaic_6849-3125-abneg   \n",
       "\n",
       "                                                                                            title  \\\n",
       "0           SARS-CoV-2 RBD mutational antig Illumina barcode sequencing for beta_19-26-none-0-ref   \n",
       "1           SARS-CoV-2 RBD mutational antig Illumina barcode sequencing for beta_19-26-none-0-ref   \n",
       "2  SARS-CoV-2 RBD mutational antig Illumina barcode sequencing for beta_19-mosaic_6848-3125-abneg   \n",
       "3  SARS-CoV-2 RBD mutational antig Illumina barcode sequencing for beta_19-mosaic_6848-3125-abneg   \n",
       "4  SARS-CoV-2 RBD mutational antig Illumina barcode sequencing for beta_20-mosaic_6849-3125-abneg   \n",
       "\n",
       "  library_strategy library_source library_selection library_layout  platform  \\\n",
       "0         AMPLICON      SYNTHETIC               PCR         single  ILLUMINA   \n",
       "1         AMPLICON      SYNTHETIC               PCR         single  ILLUMINA   \n",
       "2         AMPLICON      SYNTHETIC               PCR         single  ILLUMINA   \n",
       "3         AMPLICON      SYNTHETIC               PCR         single  ILLUMINA   \n",
       "4         AMPLICON      SYNTHETIC               PCR         single  ILLUMINA   \n",
       "\n",
       "      instrument_model                 design_description filetype  \\\n",
       "0  Illumina HiSeq 2500  PCR of barcodes from RBD variants    fastq   \n",
       "1  Illumina HiSeq 2500  PCR of barcodes from RBD variants    fastq   \n",
       "2  Illumina HiSeq 2500  PCR of barcodes from RBD variants    fastq   \n",
       "3  Illumina HiSeq 2500  PCR of barcodes from RBD variants    fastq   \n",
       "4  Illumina HiSeq 2500  PCR of barcodes from RBD variants    fastq   \n",
       "\n",
       "                                   filename  \\\n",
       "0  beta_19-26_lib1_ref-3_S3_R1_001.fastq.gz   \n",
       "1  beta_19-26_lib2_ref-2_S6_R1_001.fastq.gz   \n",
       "2     beta_19_lib1_abneg_S9_R1_001.fastq.gz   \n",
       "3    beta_19_lib2_abneg_S17_R1_001.fastq.gz   \n",
       "4    beta_20_lib1_abneg_S10_R1_001.fastq.gz   \n",
       "\n",
       "                                  filename2  \\\n",
       "0  beta_19-26_lib1_ref-2_S2_R1_001.fastq.gz   \n",
       "1  beta_19-26_lib2_ref-4_S8_R1_001.fastq.gz   \n",
       "2                                       NaN   \n",
       "3                                       NaN   \n",
       "4                                       NaN   \n",
       "\n",
       "                                  filename3  \\\n",
       "0  beta_19-26_lib1_ref-4_S4_R1_001.fastq.gz   \n",
       "1  beta_19-26_lib2_ref-1_S5_R1_001.fastq.gz   \n",
       "2                                       NaN   \n",
       "3                                       NaN   \n",
       "4                                       NaN   \n",
       "\n",
       "                                  filename4  \n",
       "0  beta_19-26_lib1_ref-1_S1_R1_001.fastq.gz  \n",
       "1  beta_19-26_lib2_ref-3_S7_R1_001.fastq.gz  \n",
       "2                                       NaN  \n",
       "3                                       NaN  \n",
       "4                                       NaN  "
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "submissions_wide = (\n",
    "    submissions_tidy\n",
    "    .assign(\n",
    "        filename_count=lambda x: x.groupby(['biosample_accession', 'library_ID'])['filename'].cumcount() + 1,\n",
    "        filename_col=lambda x: 'filename' + x['filename_count'].map(lambda c: str(c) if c > 1 else '')\n",
    "        )\n",
    "    .pivot(\n",
    "        index='library_ID',\n",
    "        columns='filename_col',\n",
    "        values='filename',\n",
    "        )\n",
    "    )\n",
    "\n",
    "submissions_wide = (\n",
    "    submissions_tidy\n",
    "    .drop(columns=['filename_fullpath', 'filename'])\n",
    "    .drop_duplicates()\n",
    "    .merge(submissions_wide[natsort.natsorted(submissions_wide.columns)],\n",
    "           on='library_ID',\n",
    "           validate='one_to_one',\n",
    "           )\n",
    "    )\n",
    "\n",
    "print('Here are the first few submission entries in wide format:')\n",
    "submissions_wide.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now write the wide submissions data frame to a `*.tsv` file that can be used for uploading the SRA submission website:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "submissions_spreadsheet = 'SRA_submission_spreadsheet.tsv'\n",
    "\n",
    "submissions_wide.to_csv(submissions_spreadsheet, sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We also want to write a file with all of the FASTQ files that are in the submission spreadsheet:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Writing list of all the FASTQ files to upload to FASTQs_to_upload.csv\n"
     ]
    }
   ],
   "source": [
    "fastq_file_list = 'FASTQs_to_upload.csv'\n",
    "\n",
    "print(f\"Writing list of all the FASTQ files to upload to {fastq_file_list}\")\n",
    "\n",
    "(submissions_tidy\n",
    " [['filename_fullpath', 'filename']]\n",
    " .to_csv(fastq_file_list, index=False)\n",
    " )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "language_info": {
   "name": "python"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
